[Shell] Provide a stack trace for panics occurring when evaluating a snippet

Consider the following sequence of commands:
```
=$ function foo() {                                                                                                                                                                                                
 > int x = 1/0;
 > }

=$ function bar() {                                                                                                                                                                                                 
 > foo();
 > }

=$ bar()
panic: {ballerina}DivisionByZero {"message":" / by zero"}
| Execution aborted due to unhandled runtime error.
```
As it can be seen, this doesn't contain a stack trace. Hence, it's impossible to identify from which component this panic originated. It would be good if we can provide a stack trace for these as well.

--- a/ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java
+++ b/ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java
@@ -434,7 +434,7 @@ public abstract class ShellSnippetsInvoker extends DiagnosticReporter {
             }
         } catch (InvokerPanicException panicError) {
             List<String> stacktrace = Arrays.stream(panicError.getCause().getStackTrace())
-                    .filter(element -> !(element.toString().contains(MODULE_STATEMENT_METHOD_NAME)||
+                    .filter(element -> !(element.toString().contains(MODULE_STATEMENT_METHOD_NAME) ||
                                         element.toString().contains(MODULE_RUN_METHOD_NAME)))
                     .collect(Collectors.toList())
                     .stream().map(element -> "at " + element.getMethodName() + "()").collect(Collectors.toList());

--- a/ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java
+++ b/ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java
@@ -64,12 +64,14 @@ import java.nio.file.Path;
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
 /**
  * Invoker that invokes a command to evaluate a list of snippets.
@@ -86,6 +88,8 @@ import java.util.function.Function;
 public abstract class ShellSnippetsInvoker extends DiagnosticReporter {
     /* Constants related to execution */
     protected static final String MODULE_RUN_METHOD_NAME = "__run";
+
+    protected static final String MODULE_STATEMENT_METHOD_NAME = "__stmt";
     private static final String MODULE_INIT_CLASS_NAME = "$_init";
     private static final String CONFIGURE_INIT_CLASS_NAME = "$configurationMapper";
     private static final String MODULE_INIT_METHOD_NAME = "$moduleInit";
@@ -429,7 +433,13 @@ public abstract class ShellSnippetsInvoker extends DiagnosticReporter {
                 errorStream.println("fail: " + failErrorMessage);
             }
         } catch (InvokerPanicException panicError) {
+            List<String> stacktrace = Arrays.stream(panicError.getCause().getStackTrace())
+                    .filter(element -> !(element.toString().contains(MODULE_STATEMENT_METHOD_NAME)||
+                                        element.toString().contains(MODULE_RUN_METHOD_NAME)))
+                    .collect(Collectors.toList())
+                    .stream().map(element -> "at " + element.getMethodName() + "()").collect(Collectors.toList());
             errorStream.println("panic: " + StringUtils.getErrorStringValue(panicError.getCause()));
+            stacktrace.forEach(errorStream::println);
             addErrorDiagnostic("Execution aborted due to unhandled runtime error.");
             throw panicError;
         } catch (MalformedURLException e) {

--- a/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/errors.checkpanic.json
+++ b/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/errors.checkpanic.json
@@ -19,7 +19,7 @@
   {
     "description": "Check panic on an error.",
     "code": "int z = checkpanic parse(\"Invalid\");",
-    "stdout": "panic: {ballerina/lang.int}NumberParsingError {\"message\":\"'string' value 'Invalid' cannot be converted to 'int'\"}\n",
+    "stdout": "panic: {ballerina/lang.int}NumberParsingError {\"message\":\"'string' value 'Invalid' cannot be converted to 'int'\"}\nat fromString()\nat parse()\n",
     "error": "InvokerPanicException"
   }
 ]

--- a/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/errors.panic.json
+++ b/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/errors.panic.json
@@ -23,7 +23,7 @@
   {
     "description": "View error.",
     "code": "readRecord(r2);",
-    "stdout": "panic: Record is nil {}\n",
+    "stdout": "panic: Record is nil {}\nat readRecord()\n",
     "error": "InvokerPanicException"
   }
 ]

--- a/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/operations.immutable.json
+++ b/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/operations.immutable.json
@@ -56,7 +56,7 @@
   {
     "description": "Cannot update immutables.",
     "code": "addEntryToMap(m2, \"intValTwo\", 10);",
-    "stdout": "panic: {ballerina/lang.map}InvalidUpdate {\"message\":\"Invalid map insertion: modification not allowed on readonly value\"}\n",
+    "stdout": "panic: {ballerina/lang.map}InvalidUpdate {\"message\":\"Invalid map insertion: modification not allowed on readonly value\"}\nat addEntryToMap()\n",
     "error": "InvokerPanicException"
   },
   {

--- a/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/regression.panic.save.state.json
+++ b/ballerina-shell/modules/shell-core/src/test/resources/testcases/evaluator/regression.panic.save.state.json
@@ -10,7 +10,7 @@
   {
     "description": "Call function f",
     "code": "f()",
-    "stdout": "panic: Error {}\n",
+    "stdout": "panic: Error {}\nat f()\n",
     "error": "InvokerPanicException"
   },
   {

--- a/misc/ls-extensions/modules/bal-shell-service/src/test/resources/testcases/getResult/errors.check.json
+++ b/misc/ls-extensions/modules/bal-shell-service/src/test/resources/testcases/getResult/errors.check.json
@@ -37,7 +37,7 @@
       "errors":[],
       "diagnostics":["Execution aborted due to unhandled runtime error."],
       "metaInfo":{"definedVars":[],"moduleDclns":[]},
-      "consoleOut":"panic: {ballerina/lang.int}NumberParsingError {\"message\":\"\u0027string\u0027 value \u002712.12\u0027 cannot be converted to \u0027int\u0027\"}"
+      "consoleOut":"panic: {ballerina/lang.int}NumberParsingError {\"message\":\"\u0027string\u0027 value \u002712.12\u0027 cannot be converted to \u0027int\u0027\"}\nat fromString()\nat parse()"
     }
   }
 ]

--- a/misc/ls-extensions/modules/bal-shell-service/src/test/resources/testcases/getResult/errors.checkpanic.json
+++ b/misc/ls-extensions/modules/bal-shell-service/src/test/resources/testcases/getResult/errors.checkpanic.json
@@ -37,7 +37,7 @@
       "errors":[],
       "diagnostics":["Execution aborted due to unhandled runtime error."],
       "metaInfo":{"definedVars":[],"moduleDclns":[]},
-      "consoleOut":"panic: {ballerina/lang.int}NumberParsingError {\"message\":\"\u0027string\u0027 value \u002712.12\u0027 cannot be converted to \u0027int\u0027\"}"
+      "consoleOut":"panic: {ballerina/lang.int}NumberParsingError {\"message\":\"\u0027string\u0027 value \u002712.12\u0027 cannot be converted to \u0027int\u0027\"}\nat fromString()\nat parse()"
     }
   }
 ]

